pkg/stanza/operator/input/windows/input.go

```diff
@@ -20,19 +23,80 @@ import (
 // Input is an operator that creates entries using the windows event log api.
 type Input struct {
 	helper.InputOperator
-	bookmark         Bookmark
-	subscription     Subscription
-	buffer           Buffer
-	channel          string
-	maxReads         int
-	startAt          string
-	raw              bool
-	excludeProviders []string
-	pollInterval     time.Duration
-	persister        operator.Persister
-	publisherCache   publisherCache
-	cancel           context.CancelFunc
-	wg               sync.WaitGroup
+	bookmark            Bookmark
+	buffer              Buffer
+	channel             string
+	maxReads            int
+	startAt             string
+	raw                 bool
+	excludeProviders    []string
+	pollInterval        time.Duration
+	persister           operator.Persister
+	publisherCache      publisherCache
+	cancel              context.CancelFunc
+	wg                  sync.WaitGroup
+	subscription        Subscription
+	remote              RemoteConfig
+	remoteSessionHandle windows.Handle
+	startRemoteSession  func() error
+}
+
+// NewInput creates a new Input operator.
+func NewInput(logger *zap.Logger, settings component.TelemetrySettings) *Input {
+	basicConfig := helper.NewBasicConfig("windowseventlog", "input")
+	basicOperator, _ := basicConfig.Build(settings)
+
+	input := &Input{
+		InputOperator: helper.InputOperator{
+			WriterOperator: helper.WriterOperator{
+				BasicOperator: basicOperator,
+			},
+		},
+	}
+	input.startRemoteSession = input.defaultStartRemoteSession
+	return input
+}
+
+// defaultStartRemoteSession starts a remote session for reading event logs from a remote server.
+func (i *Input) defaultStartRemoteSession() error {
+	if i.remote.Server == "" {
+		return nil
+	}
+
+	login := EvtRpcLogin{
+		Server:   windows.StringToUTF16Ptr(i.remote.Server),
+		User:     windows.StringToUTF16Ptr(i.remote.Username),
+		Password: windows.StringToUTF16Ptr(i.remote.Password),
+	}
+
+	sessionHandle, err := evtOpenSession(EvtRpcLoginClass, &login, 0, 0)
+	if err != nil {
+		return fmt.Errorf("failed to open session for server %s: %w", i.remote.Server, err)
+	}
+	i.remoteSessionHandle = sessionHandle
+	return nil
+}
+
+// stopRemoteSession stops the remote session if it is active.
+func (i *Input) stopRemoteSession() error {
+	if i.remoteSessionHandle != 0 {
+		if err := evtClose(uintptr(i.remoteSessionHandle)); err != nil {
+			i.Logger().Error("Failed to close remote session handle", zap.String("server", i.remote.Server), zap.Error(err))
```

Comment:
Similar to previous comment possible double logging.

Reply:
removed double logging

Reply:
This one is still here :)